libraries/ms-common/src/stm32f0xx/soft_timer.c

```diff
@@ -203,7 +212,8 @@ static void prv_update_timer(void) {
   // and reenable compares. In the case where there aren't any timers registered, the compare
   // channel is disabled until a new timer is added.
   if (s_timers.head != NULL) {
-    TIM_SetCompare1(TIM2, s_timers.head->expiry_us);
+    TIM_SetCompare1(TIM2,
```

Comment:
I don't think this handles rollover properly?

Reply:
I fixed it but please take a look at the logic just to double check.